### PR TITLE
doc: update doc for casm = true in scarb.toml

### LINKED ISSUE #2472: Document that `casm = true` in Scarb.toml slows the compilation
### Which components does the task require to be changed? (think hard pls)

snforge, sncast

### Description

We compile every sierra with USC anyways since Scarb gives no guarantee about CASM by design.
The overhead generated by this can be huge for big sierra programs (couple dozens of MB). It was only mentioned changelog of 0.23.0 version.  It should be done in this file [here](https://github.com/foundry-rs/starknet-foundry/blob/master/docs/src/appendix/scarb-toml.md)

### PR DESCRIPTION
**Pull request by NueloSE**
Closes #2472 

## Introduced changes
Add a description about the slow compilation when casm = true in scarb.toml file

-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by cptartur — CHANGES_REQUESTED** [review]

File: docs/src/appendix/scarb-toml.md [hunk 1]
```diff
@@ -166,7 +166,7 @@ sierra = true
 
 #### `casm`
 
-See more about [CASM contract class generation](https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html#casm-contract-class-generation) in Scarb documentation.
+Enabling `casm = true` in Scarb.toml causes unnecessary overhead. Tools like `snforge` and `sncast` recompile Sierra to CASM separately, resulting in redundant processing. This duplicates CASM generation, significantly impacting performance, especially for large Sierra programs. It increases build size by several megabytes and slows down compilation. To avoid this, leave `casm = false`(the default) and let tools handle CASM generation only when needed. The Sequencer performs CASM compilation for execution, making local generation redundant unless required for offline verification or specific development workflows. See more about [CASM contract class generation](https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html#casm-contract-class-generation) in Scarb documentation.
```
[the remark this answers is not in the document]
Comment (NueloSE) [reply docs/src/appendix/scarb-toml.md]: resolved. Thanks

File: docs/src/appendix/scarb-toml.md [hunk 2]
```diff
@@ -158,14 +185,17 @@ The `starknet-contract` target allows to build the package as a Starknet Contrac
...
 See more about [Sierra contract class generation](https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html#sierra-contract-class-generation) in Scarb documentation.
+
 ```toml
 [[target.starknet-contract]]
 sierra = true
 ```
 
 #### `casm`
 
+Enabling [`casm = true`](https://github.com/foundry-rs/starknet-foundry/releases/tag/v0.23.0) in Scarb.toml causes unnecessary overhead and should be disabled unless required by other tools. Tools like `snforge` and `sncast` recompile Sierra to CASM separately, resulting in redundant processing. This duplicates CASM generation, significantly impacting performance, especially for large Sierra programs.
```
Comment (cptartur) [inline docs/src/appendix/scarb-toml.md:198]: I don't think we need this link

```suggestion
Enabling `casm = true` in Scarb.toml causes unnecessary overhead and should be disabled unless required by other tools. Tools like `snforge` and `sncast` recompile Sierra to CASM separately, resulting in redundant processing. This duplicates CASM generation, significantly impacting performance, especially for large Sierra programs.
```

Comment (cptartur) [inline docs/src/appendix/scarb-toml.md:198]:
```suggestion
Enabling [`casm = true`](https://github.com/foundry-rs/starknet-foundry/releases/tag/v0.23.0) in Scarb.toml causes unnecessary overhead and should be disabled unless required by other tools. Starknet Foundry compiles Sierra to CASM separately, resulting in redundant processing. This duplicates CASM generation, significantly impacting performance, especially for large Sierra programs.
```

File: docs/src/appendix/scarb-toml.md [hunk 3]
```diff
@@ -5,13 +5,16 @@ The `Scarb.toml` contains the package manifest that is needed in package compila
 ## `Scarb.toml` Contents
 
 ### `[tool.snforge]`
+
 ```toml
 [tool.snforge]
 # ...
 ```
+
```
Comment (ksew1) [inline docs/src/appendix/scarb-toml.md:13]: Why did you added `enter` in all of this places? 🤔

**Review by cptartur — COMMENTED** [review]
Otherwise looks good to me

File: docs/src/appendix/scarb-toml.md (hunk 2 above)
Comment (NueloSE) [reply docs/src/appendix/scarb-toml.md]: Hello @cptartur  and @ksew1,

Thank you both for your feedback. @cptartur, I have committed your suggestion. I noticed that your final suggestion still includes a link to v0.23.0. Could you confirm if it’s fine to keep this link, or would you prefer I remove it?

**Comment by NueloSE** [comment]
Hello @cptartur and @ksew1,

Thank you both for your feedback. @cptartur, I have committed your suggestion. I noticed that your final suggestion still includes a link to v0.23.0. Could you confirm if it’s fine to keep this link, or would you prefer I remove it?

File: docs/src/appendix/scarb-toml.md [hunk 4]
```diff
@@ -158,14 +185,17 @@ The `starknet-contract` target allows to build the package as a Starknet Contrac
...
 See more about [Sierra contract class generation](https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html#sierra-contract-class-generation) in Scarb documentation.
+
 ```toml
 [[target.starknet-contract]]
 sierra = true
 ```
 
 #### `casm`
 
+Enabling [`casm = true`](https://github.com/foundry-rs/starknet-foundry/releases/tag/v0.23.0) in Scarb.toml causes unnecessary overhead and should be disabled unless required by other tools. Starknet Foundry compiles Sierra to CASM separately, resulting in redundant processing. This duplicates CASM generation, significantly impacting performance, especially for large Sierra programs.
```
Comment (franciszekjob) [inline docs/src/appendix/scarb-toml.md:198]: @NueloSE changes mentioned in https://github.com/foundry-rs/starknet-foundry/pull/2619#discussion_r1819023130 are still not applied. Please remove the link.

**Review by franciszekjob — CHANGES_REQUESTED** [review]
Please unnecessarily added `enter`s.

**Comment by NueloSE** [comment]
@franciszekjob @cptartur @ksew1 all `enter`s have been removed. It was as a result of prettier. Apologies for the issues. Ready for review.

**Review by ksew1 — APPROVED** [review]

**Review by cptartur — APPROVED** [review]

**Pull request merged**